### DISCUSSION TITLE
Could you provide a roadmap for further development?

### DISCUSSION THREAD

**Discussion by noseshimself** [slackhq/nebula#779] [General]
Right now the central administration of nodes is repetitive work in a terminal and the config files have to be edited by hand.  Is there any planning for additional tools which might even include automatic generation of config files from templates?

**Comment by johnmaguire** [comment]
Hi @noseshimself -

There are no plans to build such tools into Nebula OSS at this time. Most users seem to deploy Nebula through a combination of tools like Ansible and Consul (with the help of something like consul-template.)

There are a couple of open-source projects I'm aware of that aim to improve this experience, e.g. https://github.com/b177y/starship and https://github.com/unreality/nebula-windows-ui.

And of course there is Defined Networking (https://defined.net) which offers a managed solution to Nebula, including CA rotation, IP address allocation, and config distribution.

To answer the question in the topic of the thread, the main goal of Nebula OSS is to remain stable and performant. The best way to get an idea of upcoming work would be to take a look at the upcoming milestones here on Github: https://github.com/slackhq/nebula/milestones

**Reply by noseshimself** [reply]
> There are a couple of open-source projects I'm aware of that aim to improve this experience, e.g. https://github.com/b177y/starship and https://github.com/unreality/nebula-windows-ui.

I probably have found them all and none show any changes in the last 12 months. With Nebula still being a moving target front ends have to be kept in sync. I was hoping one of these projects would be adopted and kept in sync. But they are if used the wrong way a possible security risk, too...

I'll have to think hard about a secure sweets dispenser for configurations that fits my needs.

Thanks for the great work. There are other solutions out there but they are so complicated that you need elaborate docs (which they don't provide) so your design choices really delivered the best result.
